### PR TITLE
Task/compose docs WIP

### PR DESCRIPTION
**Pull request by TGazica**
### 🎯 Goal

_Describe why we are making this change_

### 🛠 Implementation details

_Describe the implementation_

### 🎨 UI Changes

_Add relevant screenshots_

| Before | After |
| --- | --- |
| img | img |

_Add relevant videos_

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="" controls="controls" muted="muted" />
</td>
<td>
<video src="" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing

_Explain how this change can be tested (or why it can't be tested)_

_Provide a patch below if it is necessary for testing_

<details>

<summary>Provide the patch summary here</summary>

```
Provide the patch code here
```

</details>

#### General
- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (required internally)
- [ ] PR targets the `develop` branch
- [ ] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)
- [ ] Tutorial starter kit updated
- [ ] Examples/guides starter kits updated (`stream-video-examples`)

### ☑️Reviewer Checklist
- [ ] XML sample runs & works
- [ ] Compose sample runs & works
- [ ] Tutorial starter kit
- [ ] Example starter kits work
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

File: docusaurus/docs/Android/03-ui/03-compose/02-abstract-compose-call-activity.mdx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    return CallViewModelFactory(
+        streamVideo = getStreamVideo(this),
+        permissionManager = PermissionManagerImpl(applicationContext),
+        usersProvider = if (BuildConfig.DEBUG) FakeUsersProvider() else EmptyUsersProvider
+    )
+}
+```
+
+<!-- TODO update after https://github.com/GetStream/stream-video-android/pull/110 has been sorted out -->
+## Customising permission manager
```
Comment (filbabic) [inline docusaurus/docs/Android/03-ui/03-compose/02-abstract-compose-call-activity.mdx]: We can also add a section on how PiP works and customizing the PiP UI - as users have to enable PiP in their manifest for it to work

File: docusaurus/docs/Android/03-ui/03-compose/02-components/01-call-content.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+The second option is the `OutgoingCall`, which is a stateless component. It allows you to provide and control the state directly.
+
+<!-- After https://github.com/GetStream/stream-video-android/pull/110 has been merged in there will be only one onCallAction handler -->
+```kotlin
+@Composable
+public fun OutgoingCall(
+    callType: CallType,
+    participants: List<CallUser>,
+    onCallAction: (CallAction) -> Unit,
```
Comment (filbabic) [inline docusaurus/docs/Android/03-ui/03-compose/02-components/01-call-content.mdx]: I think in this case the action can go last to enable the trailing lambda syntax

File: docusaurus/docs/Android/03-ui/03-compose/02-components/01-call-content.mdx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+The second option is the `IncomingCall`, which is a stateless component. It allows you to provide and control the state directly.
+
+<!-- After https://github.com/GetStream/stream-video-android/pull/110 has been merged in there will be only one onCallAction handler -->
+```kotlin
+@Composable
+public fun IncomingCall(
+    callType: CallType,
+    participants: List<CallUser>,
+    onCallAction: (CallAction) -> Unit,
```
Comment (filbabic) [inline docusaurus/docs/Android/03-ui/03-compose/02-components/01-call-content.mdx]: same here

File: docusaurus/docs/Android/03-ui/03-compose/02-components/01-call-content.mdx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+```
+
+Both allow you to override the `onCallAction` handler and implement custom logic when the user interacts with certain parts of the Call UI.
+
+<!-- TODO add some incoming call image -->
+[IncomingCallContent]()
+
+## ActiveCall
+
+To show an active call with all of its participants there are two options. The first on is `ActiveCallContent` which is a bound component. By default its state and  handlers will depend on the `CallViewModel` which can be overridden.
```
Comment (filbabic) [inline docusaurus/docs/Android/03-ui/03-compose/02-components/01-call-content.mdx]: I would maybe rephrase these to not sound too repetitive, e.g. you can use this sentence for the first component, then for the second you can say:

The <second> component behaves the same as <first>, but shows different UI and a different call state. You can use the component in the same two ways:

<then list out code and stuff>.

---
And then for the third:

Finally, our <third> component is more of the same, but represents the Active call state, when the user is in an ongoing call. The bound version is the following:

<some code and explanation>

and the stateless version:

<stateless>

Both allow you to override....
---

While semantically conveying the same information, it's not repetitive and reads better (at least from my POV).

File: stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/call/activecall/ActiveCallContent.kt [hunk 5]
```diff
@@ -48,43 +50,74 @@ import io.getstream.video.android.viewmodel.CallViewModel
...
-        false
-    )
-
-    val callMediaState by callViewModel.callMediaState.collectAsState(initial = CallMediaState())
+    onParticipantsMenuClick: () -> Unit = callViewModel::showParticipants,
+    onBackPressed: () -> Unit = {
+        val isShowingParticipantsInfo = callViewModel.isShowingParticipantsInfo.value
+        val isShowingAudioDevicePicker = callViewModel.isShowingAudioDevicePicker.value
 
-    BackHandler {
```
Comment (filbabic) [inline stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/call/activecall/ActiveCallContent.kt]: So the thing about having a back handler is that is consumes teh system back button. 

If our users use this component as a standalone, now it will no longer handle that case, only our UI back button presses.

I think we shouldn't remove this. Maybe what we can is expose a `backHandler` of the type @Composable () -> Unit that handles the system back and then have a deafult value.

Then we can send it down from the CallContent all the way down to ActiveCallContent.

LMK what you think!

File: docusaurus/docs/Android/03-ui/03-compose/02-abstract-compose-call-activity.mdx (hunk 1 above)
Comment (TGazica) [reply docusaurus/docs/Android/03-ui/03-compose/02-abstract-compose-call-activity.mdx]: Yup, makes perfect sense, i will add a title, you can fill it out when PiP is done.

File: stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/call/activecall/ActiveCallContent.kt (hunk 5 above)
Comment (TGazica) [reply stream-video-android-compose/src/main/kotlin/io/getstream/video/android/compose/ui/components/call/activecall/ActiveCallContent.kt]: If the users use the `ActiveCallContent` it will still have the usecase since we have a `onBackPressed` handler that decides what happenes on back actions, maybe should have been named better.

But yes, we can expose a `backHandler` and let the integrators implement their own solution.

File: stream-video-android/src/main/kotlin/io/getstream/video/android/viewmodel/CallViewModel.kt [hunk 6]
```diff
@@ -203,7 +203,9 @@ public class CallViewModel(
                     }
                     is State.Incoming -> {
                         _callType.value = CallType.fromType(state.callGuid.type)
-                        _participants.value = state.users.values.toList()
+                        _participants.value = state.users.values
+                            .filter { it.id != streamVideo.getUser().id }
```
Comment (filbabic) [inline stream-video-android/src/main/kotlin/io/getstream/video/android/viewmodel/CallViewModel.kt:207]: Any reason why we're filtering ourselves out from the list of participants? We _are_ participants in a call, this could easily disrupt the logic
